### ISSUE TITLE
Parent Control - Access to Addons 

### ISSUE THREAD

**Issue by tfouto** [Stremio/stremio-features#574]
It's important to implement a parental control system to prevent the installation of adult-oriented add-ons, ensuring that children are unable to access inappropriate content. This safety measure would help maintain a child-friendly environment on the platform.

To enhance the security, access to the add-ons should be secured with a PIN set through the parental control settings. Only those with the PIN can install or manage add-ons.

**Comment by victorgveloso** [comment]
In addition to that, parental control should have optional fields for rating certificates filtering.

Cinemeta should enforce the parental control feature by showing only results for content that matches the selected filters.

Both IMDB and TMDB support certifications in their API. 

TMDB: 

- https://developer.themoviedb.org/reference/certification-movie-list

IMDB:

- https://developer.imdb.com/documentation/api-documentation/sample-queries/title-name/?ref_=side_nav
- https://help.imdb.com/article/contribution/titles/certificates/GU757M8ZJ9ZPXB39?ref_=helpart_nav_27#

**Comment by makapav** [comment]
We would appreciate a method to 'prevent' adult or adult-typish content from surfacing in the search results (and Cinemata cards).

I understand that such filtering is not an exact science but even a 80% success rate is better than nothing. I believe the voice-based search already blocks many inappropriate spoken words (I have not tested it). In either case, I am hoping something similar can be done for keyboard-based entry search results - returning nothing for them.

If that's not possible - please consider adding a 'word-based blocklist' which would provide null results. 'We' as users can maintain that list as long as it is locked behind a 'pin'.

Edit: Marked assumptions and fixed inaccurate statements.

**Comment by jaruba** [comment]
_My understanding is that in the past, Stremio has made it known that they do not want to implement profiles to prevent it from mimicking an 'easy-to-use' mainstream piracy enabling media player._

you are either extremely misinformed or you are just twisting words to get the attention of the Stremio team, what exactly gave you this "understanding"?

_it's just not possible to use this product in a living room setting with kids under 16 years of age in the house._

ofcourse not, this is why the app is rated at 16+ by goverment organisations across the globe on Play Store, the app ratings are there for a reason..

_I understand that such filtering is not an exact science but even a 80% success rate is better than nothing._

80% success rate is absolutely illegal, Stremio cannot have a "kid safe" button, show you porn by mistake and then say "oopsie", it would get sued by parents and you wouldn't have Stremio at all anymore

---

please do not make up stories about what we said, and please do not try to influence our development

it is an extremely complex problem that users keep thinking is simple to add..

if it comes, then it comes, we don't announce any future plans in advance so we won't ever tell you if we are planning this feature or not, it's a free app so u get what u get..

**Comment by makapav** [comment]
I had no such nefarious intent to get the attention or developers or surreptitiously influence development. I am very grateful for the tool that the developers have built and provided to us - so a big thank you!

My expectation for this board engagement has been driven by the Google product boards - which end up becoming more of a 'discussion forum'.

I have updated my original comment accordingly - I was misinformed through some reddit forum discussions. It is now more direct with just my ask.

i was not aware of the 16+ app rating - that is helpful - thank you. My feature request to give an option to 'block' adult content or keywords for search results still remains.

I hope my comment is within your guidelines now.

**Comment by ubeyou** [comment]
Adding pin control to the settings would be extremely helpful. It would help prevent anyone from modifying & copy the add-ons.

**Comment by kintaro1981** [comment]
+1 for the possibility to filter and lock adult content (and addons)

**Comment by kintaro1981** [comment]
Just not call it “parental control” to avoid issues but add the possibility to have a locked and filtered (some addons category* disabled, hide content/file with words in it) profile can be a good start.

*add categories to Addons so NSFW can be filter/locked/hidden

**Comment by djay** [comment]
I don't think it needs profiles. Most settop devices already have profiles with dual-app support.
it would be enough that I could create a 2nd login and 
- configure a certification filter that works for both Cinemeta and search like @victorgveloso mentioned.
- hide addon install/configure and only allow install via web. (or pin is ok). 
    - even this is not as important because install doesn't even work on chromecast I think. or mine doesn't anyway

TBH profiles is worse in a way as then you need to have extra pins to get prevent switching from a kids profile to an adult. and it's a lot more work.

**Comment by tcbl33** [comment]
Also want an option to either hide add ons, hide NSFW add ons, or require a pin to install an add on. This latter is the least desirable as I'd rather the little ones simple not see the option if the stumble to the add on page. The top 2 add ons are porn for my stremio.

**Comment by FastZet** [comment]
Adding PIN control to access and manage addons tab would suffice for the majority imho. But the implementation should hide or blur the addon page that pops up when clicking addon tab till the PIN is keyed in and entered, as the 18+ addons are visible at the top.

**Comment by tcbl33** [comment]
> Adding PIN control to access and manage addons tab would suffice for the majority imho. But the implementation should hide or blur the addon page that pops up when clicking addon tab till the PIN is keyed in and entered, as the 18+ addons are visible at the top.

Maybe even just a pin to enter the add-on section and then all is well? This can be part of the settings that the user can turn off or on.

**Comment by Scullzyy** [comment]
Any updates around parental controls? This would be a welcomed feature.

**Comment by djay** [comment]
One workaround that's working for me is that if you use kids profiles on google tv you can have stremio logged into a different  stremio account that you create for your kids. Of course there is still no way to age restrict search or listings or even show a age recommendation nicely. but I suspect thats going to be very hard because I don't think tv shows have reliable age ratings like movies do so you'd have to rely on some community rating system? But I guess if you could disable search and the cinemata lists and plugins you could just add only a plugin that gives you family content...

also Annoyingly on google tv all non-kids profiles has the same app config so you can't have different stremio accounts so stremio profiles would be nice in that case. I guess you could again workaround that by making everyone but an admin user into a kids account?

**Comment by agiusthomas** [comment]
this would be very helpful since adult addons are accessible, even if just a pin on addons 🙏🙏🙏

**Comment by Hikiru** [comment]
This is ridiculous. It’d be an easy feature to add. The first thing you see when you enter the addons tab is **chaturbate**. Why has this been ignored for so long? All we need is an access pin, nothing complex. It doesn’t even need to be labeled as parental control.

**Comment by GeneralKugelBlitz** [comment]
Why this issue hasn't been closed yet?

**Comment by DougBeney** [comment]
By default, 18+ addons and content should be hidden. It cheapens the overall product when you click Addons for the first time and immediately see things like Chaturbate and stripchat.

**Comment by kKaskak** [comment]
done 

https://stremio.com/donate check user profiles feature and the child catalogs new cinemeta config offers

also check web.stremio.com/ to configure 

<img width="1520" height="1040" alt="Image" src="https://github.com/user-attachments/assets/6fdba3df-bd22-44f4-89f1-1b6ba9f53ec6" />

<img width="1589" height="987" alt="Image" src="https://github.com/user-attachments/assets/26011312-c19d-41ed-bad0-bffd7a62915f" />
